### ISSUE TITLE
Error: Not allowed to request resource

### ISSUE THREAD

**Issue by EmnaX** [lukstei/reddit-hyped-stocks#4]
Hello,
Interesting project! I had always thought it would be a cool idea to see how Reddit correlated with the stock market so thank you for putting something like this together!

I followed the steps listed for getting this running using Docker. I am running it on Ubuntu 20.04.2 LTS with Nginx proxy.

Unfortunately when I go to the website, I get the error "Error: Not allowed to request resource". From the command line, everything looks normal:

```
vps01➜  hype : main ✘ :✖✹ ᐅ  ./run-server-docker.sh
Sending build context to Docker daemon  27.62MB
Step 1/14 : FROM node:15-buster-slim AS frontend
 ---> 0107c4fc6de9
Step 2/14 : COPY frontend /app/frontend
 ---> Using cache
 ---> 12895bebde1b
Step 3/14 : WORKDIR /app/frontend
 ---> Using cache
 ---> 76ab9601e7d6
Step 4/14 : RUN npm ci
 ---> Using cache
 ---> c4218027bb8b
Step 5/14 : RUN npm run build
 ---> Using cache
 ---> 4e5bd0037789
Step 6/14 : FROM python:3.9-slim-buster
 ---> b2b5367cdfd4
Step 7/14 : RUN mkdir -p /app/backend
 ---> Using cache
 ---> 7e10de788592
Step 8/14 : WORKDIR /app/backend
 ---> Using cache
 ---> 916dcab29446
Step 9/14 : COPY backend/requirements.txt /app/backend/
 ---> Using cache
 ---> a90bb54f59d5
Step 10/14 : RUN pip install -r requirements.txt
 ---> Using cache
 ---> 5ea7224e2e96
Step 11/14 : COPY backend /app/backend
 ---> Using cache
 ---> f3ebf8a025d3
Step 12/14 : COPY --from=frontend /app/frontend/build /app/frontend/build
 ---> Using cache
 ---> 28e755264b3e
Step 13/14 : ENTRYPOINT ["python"]
 ---> Using cache
 ---> 034669003f12
Step 14/14 : CMD ["server.py"]
 ---> Using cache
 ---> 3731d9d297ec
Successfully built 3731d9d297ec
Successfully tagged pennystocks:latest
 * Serving Flask app "server" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://0.0.0.0:5000/ (Press CTRL+C to quit)
172.17.0.1 - - [17/Mar/2021 13:15:09] "GET / HTTP/1.0" 200 -
172.17.0.1 - - [17/Mar/2021 13:15:10] "GET / HTTP/1.0" 200 -
^C%
```

My Nginx `access.log` file doesn't show anything strange for those same two requests:

```
10.0.0.202 - - [17/Mar/2021:09:15:09 -0400] "GET / HTTP/1.1" 200 1175 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.3 Safari/605.1.15"
10.0.0.202 - - [17/Mar/2021:09:15:10 -0400] "GET / HTTP/1.1" 200 1175 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.3 Safari/605.1.15"
```

Nginx doesn't have anything in the `error.log` file.

![Screen Shot 2021-03-17 at 9 10 32 AM](https://user-images.githubusercontent.com/13283601/111472754-ab61df80-8700-11eb-9726-c36a8a3606b5.png)

I would be happy to provide some more log files if you let me know where to look.

Edit:
I should add that I did copy the `data.db.example` over to `data.db` and then I ran the `run-load-data-docker.sh` script which seemed to complete successfully.

**Comment by lukstei** [comment]
To be honest, I'm not sure what the problem is here, could be a CORS issue. I think it is related with your Docker setup.
What you could try is, to set it up without Docker (as described in the Readme)

**Comment by EmnaX** [comment]
> To be honest, I'm not sure what the problem is here, could be a CORS issue. I think it is related with your Docker setup.
> What you could try is, to set it up without Docker (as described in the Readme)

Thanks for the suggestions. I'll try that out tonight as well as running it without the proxy and see if that helps.

**Comment by EmnaX** [comment]
Just tried it again without the Nginx proxy and I get the message `Error: Failed to fetch`. I'll try installing it without Docker next and see what happens.

**Comment by jeremylaratro** [comment]
> Just tried it again without the Nginx proxy and I get the message `Error: Failed to fetch`. I'll try installing it without Docker next and see what happens.

I'm also having the same issue -- I get "Error: Failed to fetch" and I am not using docker. 
Edit:
this is what I found: 
"Access to fetch at 'http://localhost:3000/scores' from origin 'http://localhost:5000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled."

I saw from doing some searching that this would probably be the issue -- unfortunately I have no idea how to implement the ACAO header.
